src/expressions/sqlAggregateExpression.ts

```diff
@@ -0,0 +1,94 @@
+/*
+ * Copyright 2016-2020 Imply Data, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import { parseSqlExpression, SqlRef, SqlExpression, SqlFunction, SqlCaseSearched } from 'druid-query-toolkit';
+import { PlywoodValue } from '../datatypes/index';
+import { SQLDialect } from '../dialect/baseDialect';
+import { ChainableExpression, Expression, ExpressionJS, ExpressionValue } from './baseExpression';
+import { Aggregate } from './mixins/aggregate';
+
+export class SqlAggregateExpression extends ChainableExpression {
+  static op = 'SqlAggregate';
+
+  static substituteFilter(sqlExpression: SqlExpression, condition: SqlExpression): SqlExpression {
```

Comment:
I'd be inclined to write a unit test for this

Reply:
Done!